dynawo/sources/Modeler/Common/DYNDelayManager.cpp

```diff
@@ -133,4 +136,31 @@ DelayManager::loadDelays(const std::vector<std::string>& values) {
   return true;
 }
 
+void
+DelayManager::setGomc(state_g* const p_glocal, size_t offset) {
+  size_t index = offset;
+
+  boost::unordered_map<size_t, Delay>::iterator it;
+  triggered_ = false;
+  for (it = delays_.begin(); it != delays_.end(); ++it) {
+    if (it->second.IsTriggered()) {
+      p_glocal[index] = ROOT_UP;
+      triggered_ = true;
+    } else {
+      p_glocal[index] = ROOT_DOWN;
+    }
+    index++;
```

Comment:
Please increment the index inside the for loop

Reply:
it is not a vector, it is a map so it is more complicated to use the index in the for loop

Reply:
for (....; ++it, ++index)

Reply:
doing it lead to confusion for me. There should be only one iteration in the loop

Reply:
It is better to have whatever is moving with the loop inside the for to avoid forgetting increasing something if some branches or break appear inside

Reply:
OK